Source/WebCore/rendering/RenderBoxModelObject.cpp

```diff
@@ -360,6 +358,18 @@ DecodingMode RenderBoxModelObject::decodingModeForImageDraw(const Image& image,
     if (bitmapImage->isAsyncDecodingEnabledForTesting())
         return DecodingMode::Asynchronous;
 
+    // Animated image case.
+    if (bitmapImage->canAnimate()) {
+        if (!(bitmapImage->shouldUseAsyncDecodingForAnimatedImages() && settings().animatedImageAsyncDecodingEnabled()))
+            return DecodingMode::Synchronous;
+
+        // First frame should be decoded according to its state in the viewport.
+        if (bitmapImage->currentFrameIndex())
+            return DecodingMode::Asynchronous;
+
+        return defaultDecodingMode();
```

Comment:
To clarify, this is a behavior change for the first frame, is that right? Whereas before, we would always sync decode the first frame of an animated image, but now we will check e.g. if it's visible in the viewport to determine whether to sync decode?

If that's right, please mention this (and any other intended behavior changes) in the commit message.

Reply:
Yes it is a behavior change. And I think it is the only behavior change in this PR. But I think it's the correct behavior. I think the old behavior was done because it was easier to do it this way. There should not be a reason for disallowing us from async decoding the first frame of an animated image.

I will add a comment about this in the commit message.